### PR TITLE
Jakarta Security 4.0

### PR DESCRIPTION
**Pull request by arjantijms**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [X] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [X] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [X] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [X] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Email to PMC
   - [x] Start release review by emailing EMO 
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/security/enterprise/jakarta.security.enterprise-api/4.0.0/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/es/jakartaee11/staged/eftl/
- [x] The URL of the compatibility certification request issue:
      https://github.com/jakartaee/security/issues/338
- [X] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by mtdelgadoa** [comment]
Hello all, 
the EMO is trying to avoid duplication of release records, so we will be providing our feedback and comments related to this release directly here.

## EMO REVIEW CHECKLIST 

EDP Review status: ONGOING

### **EMO review checklist**
PMI record: [URL](https://projects.eclipse.org/projects/ee4j.security/reviews/4.0-release-review)

_**EF Specification Process**_      
- [ ] Spec Committee Ballot completed

_**Intellectual Property Management**_      
- [ ] All project code has [copyright and license headers](https://www.eclipse.org/projects/handbook/#ip-copyright-headers) correctly applied. ** EMO will scan the code at their discretion **   
- [ ]  All distributed third-party content has been vetted by the IP Due Diligence process (i.e., IP Log has been approved)   

_**Open Source Rules of Engagement**_
- [ ] PMC approval [on the mailing list](https://github.com/eclipsefdn/emo-team/emo/-/blob/main/%7Blink%20to%20the%20message%7D)

General:
- [ ] Project is operating within the mission and scope defined in its top-level project’s charter   
- [ ]  Project is operating within the bounds of its own scope   
- [ ]  Project is operating in an open and transparent manner       
- [ ]  Overall the project is operating according to the Eclipse Development Process.   

Things to check:
- Communication channels advertised
- Advertised communication channels used
- Committers are responding to questions
- Committers are responding to issues
- Committers are responding to pull/merge/review requests

_**Branding and Trademarks**_
The following applies when the project has a custom website.
To the best of our knowledge:
- [ ]  Project content correctly uses Eclipse Foundation trademarks    
- [ ] Project content (code and documentation) does not violate trademarks owned by other organizations   

Things to check:
- [ ]  Project website uses the [project's formal name](https://www.eclipse.org/projects/handbook/#trademarks-website-name) in first and all prominent references
- [ ]   Project website includes a trademark attribution statement
- [ ] Project website footers contain all [necessary elements](https://www.eclipse.org/projects/handbook/#trademarks-website-footer)

_**Legal Documentation**_
Required files:   
- [x] License files in all repository roots   
- [x]  README       
- [x]  CONTRIBUTING (or equivalent)   

Recommended files:
- [ ] NOTICES or equivalent (you can use the [Legal Documentation generator](https://www.eclipse.org/projects/tools/documentation.php) available under committer tools)   
- [x]  CODE_OF_CONDUCT   
- [ ]  SECURITY   
 
See examples for [Security file](https://gitlab.eclipse.org/security/best-practices/-/blob/main/templates/SECURITY.md) and [Code of Conduct](https://github.com/eclipse/.github/blob/master/CODE_OF_CONDUCT.md).

Required elements:
- [ ]  ECA is referenced/described   

Recommended elements:
- [ ]  Build instructions provided   
- [ ] [Security policy](https://www.eclipse.org/security/policy.php) is described   
 
_**Metadata (PMI)**_
- [x] The formal name, e.g. "Eclipse Foo™", is used in the project title   
- [x] The formal name including appropriate marks (e.g, "™") is used in the first mention in the text of the project description, and scope   
- [x] The project description starts with a single paragraph that can serve as an executive summary   
- [x] Source code repository references are up-to-date   
- [ ] Download links and information are up-to-date (see [EF handbook for more information on how-to do this](https://www.eclipse.org/projects/handbook/#resources-downloads))   
- [x] Communication channels listed in the PMI (i.e. public mailing list, forums, etc.)

**Comment by ivargrimstad** [comment]
# Mentor Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation
  - [ ] Review summary statement to ensure it has been updated to reflect the work accomplished and does not contain proposal statements as might be found in Plan Review statement.

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

10. If a Release Review is required, the specification project team contacts the EMO to initiate the review by sending an email to emo@eclipse.org.
    (A Release Review is not required if the current release is a Service Release based on a previously successful Major or Minor
    release as indicated by a release record on the project's Releases page, e.g., the [Jakarta Servlet releases page](https://projects.eclipse.org/projects/ee4j.servlet/reviews).)
    - [ ] The specification project team requests approval by sending an email to the EMO (with cc to the PMC) that contains a link to **this** PR and a request to the PMC for approval.
          
11. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by jeanouii** [comment]
Left a small comment on the CSR for Soteria. Small detail.
The TCK ZIP content has a weird structure but not a blocker

image


there is also an old-tck/ directory in there. Not sure what it is about.

Finally, I don't see a link to the Maven entry for the API jar. It's usually under the TCK links.

File: security/4.0/_index.md [hunk 1]
```diff
@@ -26,18 +23,14 @@ The release splits focuses on evolving the API in various ways, and specifically
...
 * [Jakarta Security 4.0 Release Record](https://projects.eclipse.org/projects/ee4j.security/releases/4.0)
 
-<!--
-The following can be uncommented and version information updated as they become available.
-
 * [Jakarta Security 4.0 Specification Document](./jakarta-security-spec-4.0.pdf) (PDF)
 * [Jakarta Security 4.0 Specification Document](./jakarta-security-spec-4.0.html) (HTML)
 * [Jakarta Security 4.0 Javadoc](./apidocs)
 * [Jakarta Security 4.0 TCK](https://download.eclipse.org/jakartaee/security/4.0/jakarta-security-tck-4.0.0.zip) ([sig](https://download.eclipse.org/jakartaee/security/4.0/jakarta-security-tck-4.0.0.zip.sig), [sha](https://download.eclipse.org/jakartaee/security/4.0/jakarta-security-tck-4.0.0.zip.sha256), [pub](https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub))
--->
```
Comment (jeanouii) [inline security/4.0/_index.md:30]: Missing API Jar, isn't it?

File: security/4.0/_index.md [hunk 2]
```diff
@@ -26,18 +23,14 @@ The release splits focuses on evolving the API in various ways, and specifically
...
 
 # Compatible Implementations
 
-* Soteria (TBC)
+* Soteria 4.0.0
```
Comment (jeanouii) [inline security/4.0/_index.md:33]: A link to the project is quite usual even though the name is probably enough.

**Comment by arjantijms** [comment]
>  old-tck/ directory in there. Not sure what it is about but it was also there before.

That is an embedded version of the old tck (the javatest/ant based one). Because of a lack of resources we don't have time to convert it to the new style tests. A number of other TCKs does/did the same thing.

File: security/4.0/_index.md (hunk 2 above)
Comment (arjantijms) [reply security/4.0/_index.md]: Thanks for noticing. It is done!

File: security/4.0/_index.md (hunk 1 above)
Comment (arjantijms) [reply security/4.0/_index.md:32]: Yes, it was missing. I added it back. Thanks!

**Comment by arjantijms** [comment]
@jeanouii is everything now correct?

**Comment by ivargrimstad** [comment]
I will start the ballot now

**Pull request merged**